apps/meteor/tests/end-to-end/api/livechat/14-units.ts

```diff
@@ -409,4 +413,428 @@ import { IS_EE } from '../../../e2e/config/constants';
+				}
+			});
+	};
+
+	describe('[POST] livechat/department', () => {
+		let departmentId: string;
+		let monitor1: Awaited<ReturnType<typeof createUser>>;
+		let monitor1Credentials: Awaited<ReturnType<typeof login>>;
+		let monitor2: Awaited<ReturnType<typeof createUser>>;
+		let monitor2Credentials: Awaited<ReturnType<typeof login>>;
+		let unit: IOmnichannelBusinessUnit;
+
+		const createDepartmentInUnit = (userCredentials: Credentials, departmentName: string, unitId: string) => {
+			return request
+				.post(api('livechat/department'))
+				.set(userCredentials)
+				.send({
+					department: { name: departmentName, enabled: true, showOnOfflineForm: true, showOnRegistration: true, email: 'bla@bla' },
+					departmentUnit: { _id: unitId },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('department').that.is.an('object');
+					expect(res.body.department).to.have.property('name', departmentName);
+					expect(res.body.department).to.have.property('type', 'd');
+					expect(res.body.department).to.have.property('_id');
+					departmentId = res.body.department._id;
+				});
+		};
+
+		before(async () => {
+			monitor1 = await createUser();
+			monitor2 = await createUser();
+			await createMonitor(monitor1.username);
+			monitor1Credentials = await login(monitor1.username, password);
+			await createMonitor(monitor2.username);
+			monitor2Credentials = await login(monitor2.username, password);
+			unit = await createUnit(monitor1._id, monitor1.username, []);
+		});
+
+		after(async () => Promise.all([deleteUser(monitor1), deleteUser(monitor2), deleteUnit(unit)]));
+
+		afterEach(() => {
+			if (departmentId) {
+				return deleteDepartment(departmentId);
+			}
+		});
+
+		it('should fail creating department when providing an invalid property in the department unit object', () => {
+			return request
+				.post(api('livechat/department'))
+				.set(credentials)
+				.send({
+					department: { name: 'Fail-Test', enabled: true, showOnOfflineForm: true, showOnRegistration: true, email: 'bla@bla' },
+					departmentUnit: { invalidProperty: true },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('errorType', 'invalid-params');
+				});
+		});
+
+		it('should fail creating a department into an existing unit that a monitor does not supervise', async () => {
+			const departmentName = 'Test-Department-Unit2';
+
+			await createDepartmentInUnit(monitor2Credentials, departmentName, unit._id);
+			await testDepartmentsInUnit(unit._id, unit.name, 0);
+			await testDepartmentAncestors(departmentId, departmentName);
+		});
+
+		it('should succesfully create a department into an existing unit as a livechat manager', async () => {
```

Comment:
Shouldn't we create a case for the `admin`s also?

Reply:
Sorry, the tests here were actually using an admin (even though I wrote livechat managers in the description :man_facepalming: )
I fixed all the descriptions (to mention admin instead). Since admins and livechat managers behave exactly the same in this flow, is it ok to test with only one of the roles? Or should I repeat the tests for the managers too?